launchdarkly-server-sdk/src/client.rs

```diff
@@ -207,16 +207,27 @@ impl Client {
         })
     }
 
+    /// Starts a client in the current thread, which must have a default tokio
+    /// runtime. This variant accepts a callback for tracking the time of the
+    /// last processed server event.
+    pub fn start_with_default_executor_and_callback(&self, event_processed_at: EventProcessedAt) {
```

Comment:
The idea is to use this method instead of the current `start_with_default_executor` call here

https://github.com/MaterializeInc/materialize/blob/0e5efd4f537b9a78ba71a76b2e15e0c3191b6c67/src/adapter/src/config/frontend.rs#L106

and pass a callback that updates the exposed Prometheus metric similar to this code:

https://github.com/MaterializeInc/materialize/blob/0e5efd4f537b9a78ba71a76b2e15e0c3191b6c67/src/adapter/src/config/frontend.rs#L52-L61